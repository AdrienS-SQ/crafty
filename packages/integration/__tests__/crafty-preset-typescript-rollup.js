/* global describe, it, expect, jest */

const fs = require("fs");
const path = require("path");

const rimraf = require("rimraf");

const testUtils = require("../utils");

// node-forge 0.6.33 doesn't work with jest.
// but selfsigned is fixed on this version
// as we don't use it for now, we can simply mock it
// https://github.com/jfromaniello/selfsigned/issues/16
jest.mock("node-forge");

it("Works with rollup", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-typescript-rollup/compiles")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist/js/myTSBundle.min.js")).toBeTruthy();
  expect(fs.existsSync("dist/js/myTSBundle.min.js.map")).toBeTruthy();
  expect(
    fs.readFileSync("dist/js/myTSBundle.min.js").toString("utf8")
  ).toMatchSnapshot();
});

it("Deletes rollup uglify plugin using crafty.config.js", () => {
  process.chdir(
    path.join(
      __dirname,
      "../fixtures/crafty-preset-typescript-rollup/compiles-no-uglify"
    )
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist/js/myTSBundle.min.js")).toBeTruthy();
  expect(fs.existsSync("dist/js/myTSBundle.min.js.map")).toBeTruthy();
  expect(
    fs.readFileSync("dist/js/myTSBundle.min.js").toString("utf8")
  ).toMatchSnapshot();
});

it("Fails gracefully on broken markup", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-typescript-rollup/fails")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

/*
it("Lints TypeScript with rollup", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-typescript-rollup/lints")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  //TODO :: make that TS linting errors throw errors
  // Files aren't generated on failed lint
  //expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  //expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});
*/
