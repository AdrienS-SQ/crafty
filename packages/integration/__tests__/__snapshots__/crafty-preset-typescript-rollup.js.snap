// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deletes rollup uglify plugin using crafty.config.js 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/compiles-no-uglify/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
[__:__:__] Finished 'js_myTSBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Deletes rollup uglify plugin using crafty.config.js 2`] = `
"var C = /** @class */ (function () {
    function C() {
        console.log(\\"Something happens here\\");
    }
    C.prototype.someMethod = function () {
        alert(\\"yes, alert still exists\\");
    };
    return C;
}());
var D = /** @class */ (function () {
    function D() {
        console.log(\\"Something else happens here\\");
    }
    D.prototype.someMethod = function () {
        alert(\\"yes, alert still exists\\");
    };
    return D;
}());

export { C, D };
//# sourceMappingURL=myTSBundle.min.js.map
"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stderr": "[js_myTSBundle] (!) __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/fails/js/typescript.ts (14,1): syntax error TS1068 Unexpected token. A constructor, method, accessor, or property was expected.
js/typescript.ts
[__:__:__] 'js_myTSBundle' errored after ____ ms
[__:__:__] Error: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/fails/js/typescript.ts (14,1): syntax error TS1068 Unexpected token. A constructor, method, accessor, or property was expected.
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
",
}
`;

exports[`Works with rollup 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
[__:__:__] Finished 'js_myTSBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Works with rollup 2`] = `
"var C=/** @class */function(){function e(){console.log(\\"Something happens here\\")}return e.prototype.someMethod=function(){alert(\\"yes, alert still exists\\")},e}(),D=/** @class */function(){function e(){console.log(\\"Something else happens here\\")}return e.prototype.someMethod=function(){alert(\\"yes, alert still exists\\")},e}();export{C,D};
//# sourceMappingURL=myTSBundle.min.js.map
"
`;
